core/src/main/java/org/apache/iceberg/ClientPoolImpl.java

```diff
@@ -147,4 +147,8 @@ private void release(C client) {
   public int poolSize() {
     return poolSize;
   }
+
+  public boolean isClosed() {
```

Comment:
package-private is enough? For UT, we could mark it as `VisibleForTesting`

Reply:
ClientPoolImpl and UT are in different packages, so package-private is not enough.
But I will add VisibleForTesting.

cc @chenjunjiedada

Reply:
@VisibleForTesting + public cannot pass Iceberg's style check, so I use public.

Since `boolean isClosed()` is read-only function, public will not introduce risks.

Besides,  `int poolSize()` is this class is also public.